### PR TITLE
Make benchmarks configurable via JSON

### LINKED ISSUE #901: Benchmark time measures misleading
```
        {"real_time", time_per_query},
        {"cpu_time", time_per_query},
```

I would expect "real_time" to give me the walltime that benchmark took. We can include `real_time_per_iteration`.

Also, CPU time is not measured. That is a measure that should roughly be doubled when the number of threads is doubled. It would be cool to have, but unless we need it, maybe we should just leave it out?

### PR DESCRIPTION
**Pull request by lawben**
This PR builds on #876. When running a benchmark, we can now either use the command line args:

```bash
> ./build/hyriseBenchmarkTPCH --scale=0.01 --encoding=dictionary ...
```

or a JSON file, which must be the first (and only) argument (all other command line args will be ignored). The name of the config file is not important, but it must end in `.json` to be recognised

```bash
> ./build/hyriseBenchmarkTPCH config.json

> cat config.json
{
  "verbose": false,
  "queries": [1, 9],
  "scale": 0.001,
  "chunk_size": 10000,
  "encoding": "unencoded",
  "time": 10,
  "scheduler": true
}
```

Like with the command line args, all missing JSON keys will be replaced by the default values.

EDIT: Now also closes #901.

### CODE REVIEW

File: src/benchmarklib/benchmark_utils.cpp [hunk 1]
```diff
@@ -68,4 +77,140 @@ BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool
...
+bool CLIConfigParser::cli_has_json_config(const int argc, char** argv) {
+  const auto has_json = argc > 1 && boost::algorithm::ends_with(argv[1], ".json");
+  if (has_json && argc > 2) {
+      std::cout << "Passed multiple args with a json config. All CLI args will be ignored...";
+  }
+
+  return has_json;
+}
+
+nlohmann::json CLIConfigParser::config_file_to_json(const std::string& json_file_str) {
```
Comment (mrks) [inline src/benchmarklib/benchmark_utils.cpp]: This does not sound right. The config file *is* a json. Maybe `parse_json_config_file` or something alike?

File: src/benchmarklib/benchmark_utils.cpp [hunk 2]
```diff
@@ -68,4 +77,140 @@ BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool
...
+  Assert(filesystem::is_regular_file(json_file_str), "No such file " + json_file_str);
+
+  nlohmann::json json_config;
+  std::ifstream json_file{json_file_str};
+  json_file >> json_config;
+
+  return json_config;
+}
+
+BenchmarkConfig CLIConfigParser::parse_default_json_config(const nlohmann::json& json_config) {
```
Comment (mrks) [inline src/benchmarklib/benchmark_utils.cpp]: If I see a method "parse default ...", I would expect a non-default as well

File: src/benchmarklib/benchmark_utils.cpp [hunk 3]
```diff
@@ -68,4 +77,140 @@ BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool
...
+  const auto default_config = BenchmarkConfig::get_default_config();
+
+  // Should the benchmark be run in verbose mode
+  const auto verbose = json_config.value("verbose", default_config.verbose);
+  auto& out = get_out_stream(verbose);
+
+  // In non-verbose mode, disable performance warnings
```
Comment (mrks) [inline src/benchmarklib/benchmark_utils.cpp]: The warning that Hyrise was built in debug mode still appears, right?

File: src/benchmarklib/benchmark_utils.cpp [hunk 4]
```diff
@@ -68,4 +77,140 @@ BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool
...
+  } else {
+    out << "- Writing benchmark results to stdout" << std::endl;
+  }
+
+  // Display info about MVCC being enabled or not
+  const auto enable_mvcc = json_config.value("mvcc", default_config.use_mvcc == UseMvcc::Yes);
+  const auto use_mvcc = enable_mvcc ? UseMvcc::Yes : UseMvcc::No;
+  out << "- MVCC is " << (enable_mvcc ? "enabled" : "disabled") << std::endl;
+
+  // Initialise the scheduler if the benchmark was requested to run multi-threaded
```
Comment (mrks) [inline src/benchmarklib/benchmark_utils.cpp]: Naming: Initializing the scheduler goes way beyond just parsing the json object.

File: src/benchmarklib/benchmark_utils.cpp [hunk 5]
```diff
@@ -68,4 +77,140 @@ BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool
...
+  return BenchmarkConfig{
+      benchmark_mode, verbose,          chunk_size,       encoding_type,        max_runs, timeout_duration,
+      use_mvcc,       output_file_path, enable_scheduler, enable_visualization, out};
+}
+
+BenchmarkConfig CLIConfigParser::parse_default_cli_options(const cxxopts::ParseResult& parse_result) {
+  return parse_default_json_config(default_cli_options_to_json(parse_result));
+}
+
+nlohmann::json CLIConfigParser::default_cli_options_to_json(const cxxopts::ParseResult& parse_result) {
```
Comment (mrks) [inline src/benchmarklib/benchmark_utils.cpp]: Also: What is default about this?

File: src/benchmarklib/benchmark_utils.cpp (hunk 3 above)
Comment (lawben) [reply src/benchmarklib/benchmark_utils.cpp]: yes. but thanks for the pointer. there was an error here that i had to fix

**Comment by lawben** [comment]
@mrks merge?

**Pull request merged**